### ISSUE TITLE
Magento2 issues.

### ISSUE THREAD

**Issue by willatplatform** [drud/ddev-platformsh#95]
when trying to use this with platform.sh and magento we are getting the following error: 

> Package container-interop/container-interop is abandoned, you should avoid using it. Use psr/container instead.
> Package laminas/laminas-console is abandoned, you should avoid using it. Use laminas/laminas-cli instead.
> Package flow/jsonpath is abandoned, you should avoid using it. Use softcreatr/jsonpath instead.
> Package fzaninotto/faker is abandoned, you should avoid using it. No replacement was suggested.
> Package phpunit/php-token-stream is abandoned, you should avoid using it. No replacement was suggested.
> Package sebastian/finder-facade is abandoned, you should avoid using it. No replacement was suggested.
> Generating autoload files
> 88 packages you are using are looking for funding.
> Use the `composer fund` command to find out more!
> PHP CodeSniffer Config installed_paths set to ../../magento/magento-coding-standard,../../phpcompatibility/php-compatibility
>  ____  _____ ____  _     _____   ____  __ _____ _   _ _____
> |  _ \| ____|  _ \| |   / _ \ \ / /  \/  | ____| \ | |_   _|
> | | | |  _| | |_) | |  | | | \ V /| |\/| |  _| |  \| | | |
> | |_| | |___|  __/| |__| |_| || | | |  | | |___| |\  | | |
> |____/|_____|_|   |_____\___/ |_| |_|  |_|_____|_| \_| |_|
> 
>     _    ____   ___  ____ _____ _____ ____
>    / \  | __ ) / _ \|  _ \_   _| ____|  _ \
>   / _ \ |  _ \| | | | |_) || | |  _| | | | |
>  / ___ \| |_) | |_| |  _ < | | | |___| |_| |
> /_/   \_\____/ \___/|_| \_\|_| |_____|____/
> 
> ------
> 
> Cannot find the main route for Magento. Please add `id: magento_route` to your routes.yaml.
> 
> ------
> Task failed: Exec command 'set -e
> cp .versioned_backup/etc/* app/etc/
> php deploy.php
> ' in container/service 'web': exit status 1
> A task failure does not mean that ddev failed, but your hook configuration has a command that failed.
> Successfully started ece-test
> Project can be reached at https://ece-test.ddev.site https://127.0.0.1:32824

The code kinda deploys out but does not generate a lot of things that are needed mostly the env.php for database connection.

I was looking at my .platform.app.yaml and it has the following this is the basic magento template at platform nothing added or removed:

> # This file describes an application. You can have multiple applications
> # in the same project.
> #
> # See https://docs.platform.sh/user_guide/reference/platform-app-yaml.html
> 
> # The name of this app. Must be unique within a project.
> name: magento
> 
> # The runtime the application uses.
> type: php:7.4
> 
> dependencies:
>     php:
>         composer/composer: "^1"
> 
> ###
> # Specify additional PHP extensions that should be loaded.
> # To determine what you need to enable
> #   1. run locally: composer check-platform-reqs | grep "ext-\|php"
> #   2. Compare to https://docs.platform.sh/languages/php/extensions.html
> #      - "Def" = Already Enabled
> #      - "Avail" = Available, but you need to enable below.
> runtime:
>     extensions:
>         - xsl
>         - sodium
> 
> # Customize your PHP.ini.
> # See: https://docs.platform.sh/languages/php/ini.html
> variables:
>     php:
>         # Include Magento's out-of-the-box settings
>         include: /app/.user.ini
> 
> # Configuration of the build of this application.
> build:
>     flavor: composer
> 
> # The relationships of the application with services or other applications.
> #
> # The left-hand side is the name of the relationship as it will be exposed
> # to the application in the PLATFORM_RELATIONSHIPS variable. The right-hand
> # side is in the form `<service name>:<endpoint name>`.
> relationships:
>     database: magento_database:mysql
>     redis: magento_redis:redis
>     search: magento_search:opensearch
> 
> # The size of the persistent disk of the application (in MB).
> disk: 2048
> 
> # The 'mounts' describe writable, persistent filesystem mounts in the application.
> mounts:
>     "var": "shared:files/var"
>     "generated": "shared:files/generated"
>     "app/etc": "shared:files/etc"
>     "pub/media": "shared:files/media"
>     "pub/static": "shared:files/static"
> 
> # The hooks executed at various points in the lifecycle of the application.
> hooks:
>     # The build hook runs on your code to prepare it for an active environment.
>     build: |
>         set -e
>         mkdir -p .versioned_backup/etc && cp app/etc/* .versioned_backup/etc/
>     # The deploy hook runs after your application has been deployed and started.
>     deploy: |
>         set -e
>         cp .versioned_backup/etc/* app/etc/
>         php deploy.php
> 
> # The configuration of app when it is exposed to the web.
> web:
>     locations:
>         '/':
>             root: "pub"
>             passthru: "/index.php"
>             index:
>                 - index.php
>             scripts: true
>             allow: false
>             rules:
>                 \.(css|js|map|hbs|gif|jpe?g|png|tiff|wbmp|ico|jng|bmp|svgz|midi?|mp?ga|mp2|mp3|m4a|ra|weba|3gpp?|mp4|mpe?g|mpe|ogv|mov|webm|flv|mng|asx|asf|wmv|avi|ogx|swf|jar|ttf|eot|woff|otf|html?)$:
>                     allow: true
>                 ^/sitemap(.*)\.xml$:
>                     passthru: "/media/sitemap$1.xml"
>         '/media':
>             root: "pub/media"
>             allow: true
>             scripts: false
>             passthru: "/get.php"
>             expires: 1y
>         '/static':
>             root: "pub/static"
>             allow: true
>             scripts: false
>             passthru: "/static.php"
>             expires: 1y
>             rules:
>                 ^/static/version\d+/(?<resource>.*)$:
>                     passthru: "/static/$resource"
> 
> # The configuration of scheduled execution.
> crons:
>     magento:
>         spec: "*/5 * * * *"
>         cmd: "php bin/magento cron:run"
> 
> workers:
>     queue:
>         size: S
>         disk: 128
>         commands:
>             start: |
>                 bin/magento queue:consumers:start async.operations.all --single-thread --max-messages=10000

**Comment by rfay** [comment]
I believe the plain-vanilla https://github.com/platformsh-templates/magento2ce results in this problem, could you please confirm that?

**Comment by willatplatform** [comment]
correct.

**Comment by gilzow** [comment]
See: https://github.com/platformsh-templates/magento2ce/blob/master/deploy.php#L93 
and 
https://github.com/platformsh-templates/magento2ce/blob/master/.platform/routes.yaml#L2

**Comment by gilzow** [comment]
_IF I remember correctly_, the Magento installer [requires a base-url](https://github.com/platformsh-templates/magento2ce/blob/daf82ca10c0210671cebf99a4d3068ef4325b197/deploy.php#L175). Since any number of routes/domains can be "primary" routes, there needs to be a reliable way to get what is ultimately the default domain. 

@tylerssn does that sound correct? 

This isn't working with the add-on since we aren't actually parsing the routes.yaml file yet but statically creating the route information for ddev. 

The script itself is running because it checks to see if the file `app/etc/installed` exists, and if not, proceeds to attempt to install Magento. `app/etc` is a file mount: https://github.com/platformsh-templates/magento2ce/blob/master/.platform.app.yaml#L56 

In this case, you'll need to do a `ddev pull platform` and get the db and mounts before you run ddev start so the files are available locally for when ddev runs the hooks.

**Comment by tylerssn** [comment]
> IF I remember correctly, the Magento installer [requires a base-url](https://github.com/platformsh-templates/magento2ce/blob/daf82ca10c0210671cebf99a4d3068ef4325b197/deploy.php#L175). Since any number of routes/domains can be "primary" routes, there needs to be a reliable way to get what is ultimately the default domain.

That's right, we're essentially being explicit about which domain we want to use and we don't have a "give me the default domain" option (that I know of) that could replace this logic.

_**That said**_, Magento doesn't technically require that a base URL be set. If it is not set, Magento will answer a request made to it from any domain. I'm not sure what the impact of that would be on this setup script—I haven't looked at it in a while 😄

**Comment by rfay** [comment]
The "Cannot find the main route" issue is now resolved.

Now we're dependent on getting the template fixed:
* https://github.com/platformsh-templates/magento2ce/pull/54
